tests/test_data.py

```diff
@@ -1231,6 +1232,32 @@ def test_numpy_c_contiguous():
     assert(u._data_allocated.flags.c_contiguous)
 
 
+@skipif(['yask', 'ops'])
+def test_numpy_allocator():
+    shape = (2, 2)
+    space_order = 0
+    numpy_array = np.ones(shape, dtype=np.float32)
+    g = Grid(shape)
+    f = Function(name='f', space_order=space_order, grid=g,
+                 allocator=NumpyAllocator(numpy_array), initializer=lambda x: None)
```

Comment:
I assume that this `initializer = lambda x: None` is here to avoid an extra memset ? so this is basically implicit to whenever NumpyAllocator is used?

Reply:
It is more than just "extra". If the data in `numpy_array` has meaning - e.g. initial conditions for the PDE I want to solve, `fill(0.)` will erase the data that was valuable. So in this case, the memset is harmful as well as being extra. 

Yes I'm using a noop initialiser here to bypass that memset. 

I don't think anything is implicit, unless I misunderstood what you mean. 

Use the `NumpyAllocator` when you want to use externally allocated memory as the backend for a `devito.Function`. 

If the data in that memory is important, use a noop `initializer` to preserve it, else, Devito's default behaviour will be to overwrite it with 0.

Reply:
oh, OK, makes sense